rewriter/Minitest.cc

```diff
@@ -267,25 +312,86 @@ ast::ExpressionPtr runUnderEach(core::MutableContext ctx, core::NameRef eachName
             auto method = addSigVoid(ast::MK::SyntheticMethod0(send->loc, declLoc, move(name), move(each)));
             // add back any moved constants
             return constantMover.addConstantsToExpression(send->loc, move(method));
-        } else if (send->fun == core::Names::describe() && send->numPosArgs() == 1 && correctBlockArity) {
+        } else if ((isDescribeOrSimilar(send->fun) || isSharedExamples(send->fun)) && send->numPosArgs() == 1 &&
+                   correctBlockArity) {
             return prepareTestEachBody(ctx, eachName, std::move(send->block()->body), args, destructuringStmts,
                                        iteratee,
                                        /* insideDescribe */ true);
-        } else if (insideDescribe && send->fun == core::Names::let() && send->numPosArgs() == 1 && correctBlockArity &&
-                   ast::isa_tree<ast::Literal>(send->getPosArg(0))) {
-            auto argLiteral = ast::cast_tree_nonnull<ast::Literal>(send->getPosArg(0));
-            if (argLiteral.isName()) {
-                auto declLoc = send->loc.copyWithZeroLength().join(argLiteral.loc);
-                auto methodName = argLiteral.asName();
-                return ast::MK::SyntheticMethod0(send->loc, declLoc, methodName, std::move(send->block()->body));
+        } else if (insideDescribe && send->fun == core::Names::itBehavesLike() && send->numPosArgs() == 1) {
+            // Handle it_behaves_like within test_each blocks
+            auto &arg = send->getPosArg(0);
+            auto argString = to_s(ctx, arg);
+            // Create an include statement that includes the shared_examples companion module
+            auto moduleName = ast::MK::UnresolvedConstant(
+                arg.loc(), ast::MK::EmptyTree(),
+                ctx.state.enterNameConstant("<shared_examples_module '" + argString + "'>"));
+            // Create an include statement to include the shared_examples companion module
+            return ast::MK::Send1(send->loc, ast::MK::Self(send->loc), core::Names::include(), send->funLoc,
+                                  std::move(moduleName));
+        } else if (insideDescribe &&
+                   ((send->fun == core::Names::let() && send->numPosArgs() == 1) ||
+                    (send->fun == core::Names::letBang() && send->numPosArgs() == 1) ||
+                    (send->fun == core::Names::subject() && send->numPosArgs() <= 1)) &&
+                   correctBlockArity && ast::isa_tree<ast::Literal>(send->getPosArg(0))) {
```

Comment:
Won't this raise if you use `subject { ... }` inside a `test_each`, because `getPosArg(0)` will be off by one?